pkg/instrumentation/kafka/headers.go

```diff
@@ -0,0 +1,48 @@
+package kafka
+
+import (
+	"github.com/twmb/franz-go/pkg/kgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type (
+	MessageCarrier struct {
+		msg *kgo.Record
+	}
+)
+
+var _ interface {
+	tracer.TextMapReader
+	tracer.TextMapWriter
+} = (*MessageCarrier)(nil)
```

Comment:
What does this block do? Are we injecting to `tracer.TextMapReader` and `tracer.TextMapWriter` our `MessageCarrier` wrapper?

Reply:
@terranisu This is done for static type checking purposes, checking that `*MessageCarrier` data type implements the `tracer.TextMapReader` and `tracer.TexMapWriter` interfaces. To learn more, check [the Effective Go](https://go.dev/doc/effective_go#blank_implements) gem